packages/core/src/modules/drpc/services/DRPCMessageService.ts

```diff
@@ -0,0 +1,122 @@
+import type { AgentContext } from '../../../agent'
+import type { InboundMessageContext } from '../../../agent/models/InboundMessageContext'
+import type { Query } from '../../../storage/StorageService'
+import type { ConnectionRecord } from '../../connections/repository/ConnectionRecord'
+import type { DRPCMessageStateChangedEvent } from '../DRPCMessageEvents'
+import type { DRPCRequest, DRPCResponse } from '../messages'
+
+import { EventEmitter } from '../../../agent/EventEmitter'
+import { injectable } from '../../../plugins'
+import { DRPCMessageEventTypes } from '../DRPCMessageEvents'
+import { DRPCMessageRole } from '../DRPCMessageRole'
+import { DRPCRequestMessage, DRPCResponseMessage } from '../messages'
+import { DRPCMessageRecord, DRPCMessageRepository } from '../repository'
+
+@injectable()
+export class DRPCMessageService {
+  private drpcMessageRepository: DRPCMessageRepository
+  private eventEmitter: EventEmitter
+
+  public constructor(drpcMessageRepository: DRPCMessageRepository, eventEmitter: EventEmitter) {
+    this.drpcMessageRepository = drpcMessageRepository
+    this.eventEmitter = eventEmitter
+  }
+
+  public async createRequestMessage(
+    agentContext: AgentContext,
+    message: DRPCRequest,
```

Comment:
In case of a batch request are we validating or automatically adding an id for each entry?

Reply:
in the agent we don't really care about the id in each DRPC request/response object. those are more for business logic to consume. So we leave it up to the business logic to validate/add ids for the DRPC messages